core/src/language/swift.rs

```diff
@@ -812,6 +787,67 @@ impl Swift {
 
         writeln!(w, "public struct CodableVoid: {} {{}}", decs.join(", "))
     }
+
+    /// Build the generic constraints output. This checks for the `swiftGenericConstraints` typeshare attribute and combines
+    /// it with the `default_generic_constraints` configuration. If no `swiftGenericConstraints` is defined then we just use
+    /// `default_generic_constraints`.
+    fn generic_constraints<'a>(
+        &'a self,
+        decorator_map: &'a DecoratorMap,
+        generic_types: &'a [String],
+    ) -> String {
+        let swift_generic_contraints_annotated = decorator_map
+            .get(&SupportedLanguage::Swift)
+            .and_then(|d| d.get(SWIFT_GENERIC_CONSTRAINTS_DECORATOR))
```

Comment:
It feels a bit awkward here (and elsewhere, especially in the `kotlin` section) to have to first get the values for a specific language, and then drill down into the decorators that match the kind we are looking for. Now that we have multiple decorators for a single language (and we already don't support decorators for _every_ language), would it be unreasonable to decouple the decorator type from the `SupportedLanguage` struct?

If we had a `DecoratorKind` enum that included the supported decorators (e.g. `Kotlin`, `Swift`, and `SwiftGenericConstraints`), and then built the map of `DecoratorKind` => `Vec` of values, the API would be roughly similar to what it is now, but allowing multiple kinds of decorators to be pulled when they are relevant.

Reply:
Yup, I agree. Pushed the update.